nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -700,28 +714,157 @@ in {
 
     services.nginx = lib.mkIf cfg.configureNginx {
       enable = true;
-      recommendedProxySettings = true; # required for redirections to work
+
+      commonHttpConfig = ''
+        proxy_cache_path /var/cache/nginx levels=1:2 keys_zone=all_cache:10m inactive=7d max_size=1g;
+      '';
+
       virtualHosts."${cfg.localDomain}" = {
         root = "${cfg.package}/public/";
         # mastodon only supports https, but you can override this if you offload tls elsewhere.
         forceSSL = lib.mkDefault true;
         enableACME = lib.mkDefault true;
 
-        locations."/system/".alias = "/var/lib/mastodon/public-system/";
-
         locations."/" = {
           tryFiles = "$uri @proxy";
+          priority = 1100;
         };
 
-        locations."@proxy" = {
-          proxyPass = (if cfg.enableUnixSocket then "http://unix:/run/mastodon-web/web.socket" else "http://127.0.0.1:${toString(cfg.webPort)}");
+        locations."= /sw.js" = {
+          tryFiles = "$uri =404";
+          priority = 2110;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=604800, must-revalidate';
+            ${nginxCommonHeaders}
+          '';
+        };
+
+        locations."^~ /assets/" = {
```

Comment:
Upstream says `~ ^/assets/`, couldn't we just use exactly that? I don't know enough about nginx locations to understan the difference, but I'd like this to be as close to upstream as possible.

Would also apply to all the other locations.

Reply:
This is the best variant:
`
~: If a tilde modifier is present, this location will be interpreted as a case-sensitive regular expression match.
^~: If a carat and tilde modifier is present, and if this block is selected as the best non-regular expression match, regular expression matching will not take place.
`

I should add a PR to Upstream, but my previous commit is still not accepted there :(